app/src/main/java/org/sagebionetworks/research/mpower/tracking/widget/TriggersLoggingUIFormItemWidget.java

```diff
@@ -0,0 +1,98 @@
+package org.sagebionetworks.research.mpower.tracking.widget;
+
+import android.content.Context;
+import android.graphics.Paint;
+import android.support.constraint.ConstraintLayout;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.sagebionetworks.research.mobile_ui.widget.ActionButton;
+import org.sagebionetworks.research.mpower.R;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class TriggersLoggingUIFormItemWidget extends ConstraintLayout {
+    @BindView(R.id.item_title)
+    TextView title;
+
+    @BindView(R.id.record_button)
+    ActionButton recordButton;
+
+    @BindView(R.id.checkmark)
+    ImageView checkmark;
+
+    @BindView(R.id.recorded_label)
+    TextView recordedLabel;
+
+    @BindView(R.id.undo_button)
+    ActionButton undoButton;
+
+    private boolean logged;
+
+    public TriggersLoggingUIFormItemWidget(final Context context) {
+        super(context);
+        this.commonInit();
+    }
+
+    public TriggersLoggingUIFormItemWidget(final Context context, final AttributeSet attrs) {
+        super(context, attrs);
+        this.commonInit();
+    }
+
+    public TriggersLoggingUIFormItemWidget(final Context context, final AttributeSet attrs, final int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        this.commonInit();
+    }
+
+    public void commonInit() {
+        inflate(this.getContext(), R.layout.mpower2_triggers_logging_item, this);
+        ButterKnife.bind(this);
+        this.undoButton.setPaintFlags(this.undoButton.getPaintFlags() | Paint.UNDERLINE_TEXT_FLAG);
```

Comment:
I'd suggest creating an UnderlinedButton class like iOS does.  That style is re-used all over the place in Woody's designs.